source/common/network/filter_manager_impl.h

```diff
@@ -11,34 +12,109 @@ namespace Envoy {
 namespace Network {
 
 /**
- * Interface used to obtain read/write buffers.
+ * Declaration of StreamBuffer backwards compatible with the former BufferSource::StreamBuffer.
  */
-class BufferSource {
+class StreamBufferSource {
 public:
-  virtual ~BufferSource() {}
+  virtual ~StreamBufferSource() {}
 
   struct StreamBuffer {
     Buffer::Instance& buffer;
     bool end_stream;
   };
+};
+
+/**
+ * Interface used to obtain read buffers.
+ */
+class ReadBufferSource : public virtual StreamBufferSource {
+public:
+  virtual ~ReadBufferSource() {}
```

Comment:
nit: not needed, same below (base class is virtual already)

Reply:
I removed `BufferSource` and  `StreamBufferSource`, so now this desctructor has to stay